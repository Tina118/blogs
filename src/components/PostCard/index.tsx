import { FC } from 'react'

interface PostCardProps {
  id: string
  name: string
  email: string
  body: string
}

const PostCard: FC<PostCardProps> = ({ id, name, email, body }) => {
  const src = 'https://source.unsplash.com/random/200x200?sig=' + id

  return (
    <div className="col-md-4 mb-3 d-flex align-item-stretch">
    <div className="card" style={{ width: '25rem' ,margin:"1rem"}}>
      <img src={src} className="card-img-top" alt="profile" width="200rem" height="200rem"/>
      <div className="card-body d-flex flex-column">
        <h5 className="card-title">{name}</h5>
        <h6 className="card-subtitle mb-2 text-muted">{email}</h6>
        <p className="card-text">{body}</p>
        <a href="#" className="btn btn-primary mt-auto align-self-start">
          Post Details
        </a>
      </div>
    </div>
    </div>
  )
}

export default PostCard
