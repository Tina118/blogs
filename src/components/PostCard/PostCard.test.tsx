import { render } from '@testing-library/react';
import PostCard from './index';

test('renders main App', () => {
  const fragment = render(<PostCard id="1" name="dummy" email="test@gmail.com"/>);
  expect(fragment).toMatchSnapshot();
});
