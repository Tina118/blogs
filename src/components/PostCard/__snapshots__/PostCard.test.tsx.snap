// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md-4 mb-3 d-flex align-item-stretch"
      >
        <div
          class="card"
          style="width: 25rem; margin: 1rem;"
        >
          <img
            alt="profile"
            class="card-img-top"
            height="200rem"
            src="https://source.unsplash.com/random/200x200?sig=1"
            width="200rem"
          />
          <div
            class="card-body d-flex flex-column"
          >
            <h5
              class="card-title"
            >
              dummy
            </h5>
            <h6
              class="card-subtitle mb-2 text-muted"
            >
              test@gmail.com
            </h6>
            <a
              class="btn btn-primary mt-auto align-self-start"
              href="posts/1"
            >
              Post Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md-4 mb-3 d-flex align-item-stretch"
    >
      <div
        class="card"
        style="width: 25rem; margin: 1rem;"
      >
        <img
          alt="profile"
          class="card-img-top"
          height="200rem"
          src="https://source.unsplash.com/random/200x200?sig=1"
          width="200rem"
        />
        <div
          class="card-body d-flex flex-column"
        >
          <h5
            class="card-title"
          >
            dummy
          </h5>
          <h6
            class="card-subtitle mb-2 text-muted"
          >
            test@gmail.com
          </h6>
          <a
            class="btn btn-primary mt-auto align-self-start"
            href="posts/1"
          >
            Post Details
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
